src/Features/Core/Portable/ConvertIfToSwitch/AbstractConvertIfToSwitchCodeRefactoringProvider.cs

```diff
@@ -1,256 +1,119 @@
-            private Optional<TStatementSyntax> _switchDefaultBodyOpt;
+            public readonly Pattern Pattern;
+            public readonly ImmutableArray<SyntaxNode> Guards;
 
-            public Analyzer(ISyntaxFactsService syntaxFacts, SemanticModel semanticModel)
+            public SwitchLabel(Pattern pattern, ImmutableArray<SyntaxNode> guards)
             {
-                _syntaxFacts = syntaxFacts;
-                _semanticModel = semanticModel;
+                Pattern = pattern;
+                Guards = guards;
             }
+        }
 
-            public async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
-            {
-                var (document, textSpan, cancellationToken) = context;
-                var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-
-                var ifStatement = await context.TryGetRelevantNodeAsync<TIfStatementSyntax>().ConfigureAwait(false);
-                if (ifStatement == null || ifStatement.ContainsDiagnostics)
-                {
-                    return;
-                }
-
-                var switchSections = GetSections(ifStatement).ToList();
-
-                // To prevent noisiness we don't offer this unless we're going to generate at least
-                // two switch labels.  It can be quite annoying to basically have this offered
-                // on pretty much any simple 'if' like "if (a == 0)" or "if (x == null)".  In these
-                // cases, the converted code just looks and feels worse, and it ends up causing the
-                // lightbulb to appear too much.
-                //
-                // This does mean that if someone has a simple if, and is about to add a lot more 
-                // cases, and says to themselves "let me convert this to a switch first!", then they'll
-                // be out of luck.  However, I believe the core value here is in taking existing large
-                // if-chains/checks and easily converting them over to a switch.  So not offering the
-                // feature on simple if-statements seems like an acceptable compromise to take to ensure
-                // the overall user experience isn't degraded.
-                var labelCount = switchSections.SelectMany(t => t.patterns).Count() +
-                    (_switchDefaultBodyOpt.HasValue ? 1 : 0);
-                if (labelCount < 2)
-                {
-                    return;
-                }
-
-                context.RegisterRefactoring(
-                    new MyCodeAction(Title,
-                        c => UpdateDocumentAsync(root, document, ifStatement, switchSections)),
-                    ifStatement.Span);
-            }
+        public sealed class SwitchSection
+        {
+            public readonly ImmutableArray<SwitchLabel> Labels;
+            public readonly IOperation Body;
+            public readonly SyntaxNode SyntaxToRemove;
```

Comment:
could we make this generic in any way.  i.e. TSwitchSection or something?  i always feel better there because then i can tell this isn't just some arbitrary node.

Reply:
this holds the node to be removed (currently ifs, throw, return) not the resulting switch section. Since VB has a handful of if-related nodes I don't think generic would be possible. I'll add a comment.

Reply:
are they always statements? or are they truly *anything*?